STY: Partially apply ruff/pycodestyle rules (E)

DimitriPapadopoulos opened the pull request:
Partially apply some ignored `E` rules.

These rules cannot be applied blindly. For example, tabular data requires extra spaces to remain aligned.

Not sure yet how to enable these rules. For example, exclude files that contain tabular data?

DimitriPapadopoulos committed:
MNT: Document ignored ruff rules

DimitriPapadopoulos committed:
STY: Partially apply ruff/pycodestyle rule E241

Multiple spaces after comma

Do not apply this rule to tabular data, tables must remain aligned.

DimitriPapadopoulos committed:
STY: Partially apply ruff/pycodestyle rule E251

Unexpected spaces around keyword / parameter equals

For now, do not modify large and consistent code blocks that use spaces on
purpose, mostly when the default values are arrays.

DimitriPapadopoulos committed:
STY: Partially apply ruff/pycodestyle rule E265

Block comment should start with `# `

For now, keep commented out code as is.

DimitriPapadopoulos committed:
STY: Partially apply ruff:pycodestyle rule E266

Too many leading `#` before block comment

For now, do not modify titles.

jorenham commented:
Maybe the mypy_primer diff is caused by this branch not being up to-date with main (or is it)?

jorenham commented on numpy/_core/tests/test_arrayprint.py:
```diff
@@ -320,7 +320,7 @@ def test_structure_format_float(self):
 
     def test_unstructured_void_repr(self):
-        a = np.array([27, 91, 50, 75,  7, 65, 10,  8,
+        a = np.array([27, 91, 50, 75,   7, 65, 10,    8,
```
🤔 
```suggestion
        a = np.array([27, 91, 50, 75, 7, 65, 10, 8,
```

DimitriPapadopoulos replied:
The idea is to have both arrays aligned. That's common practice in Numpy for tabular data.

jorenham replied:
Yea it's pretty common. But IDE that are configured to automatically `ruff format` (which is pretty common in my experience) would reformat this either way, resulting in large diffs that the reviewers will have to go through (i.e. self-interest) :P.

DimitriPapadopoulos replied:
But I thought Numpy will not be formatted with ruff any time soon, precisely because of tabular data:
* https://github.com/astral-sh/ruff/discussions/7310

If so, any patch that reformats a file using `ruff format` will be rejected anyway, won't it?

jorenham commented on numpy/ma/tests/test_core.py:
```diff
@@ -997,8 +997,8 @@ def test_mvoid_multidim_print(self):
 
-        t_ne = masked_array(data=[(1, (1, 1))],
-                            mask=[(True, (True, False))],
+        t_ne = masked_array(data = [(1, (1, 1))],
+                            mask = [(True, (True, False))],
```
🙃 

```suggestion
        t_ne = masked_array(data=[(1, (1, 1))],
                            mask=[(True, (True, False))],
```

DimitriPapadopoulos replied:
All other keyword arguments in this function use ` = ` instead of `=`. I thought consistency might be more important than fixing everything, but if we are to apply this ruff rule some day, then we need to fix all occurrences...

Fixed now.

jorenham commented on ruff.toml:
my comment at https://github.com/numpy/numpy/pull/27309#discussion_r2051567388 also applies here

DimitriPapadopoulos committed:
STY: Take into account reviewer's comments

Co-authored-by: Joren Hammudoglu <jhammudoglu@gmail.com>

DimitriPapadopoulos commented:
Rebased, but the [mypy_primer](https://github.com/hauntsaninja/mypy_primer) warning doesn't go away.

jorenham commented:
> Rebased, but the [mypy_primer](https://github.com/hauntsaninja/mypy_primer?rgh-link-date=2025-04-22T08%3A55%3A53Z) warning doesn't go away.

I honestly have no idea why there's a mypy_primer diff here...
But it probably won't matter; as I don't expect pytorch to complain that their `unused-ignore` mypy error has disappeared 🤷🏻

charris commented:
> It probably won't be enforced globally anytime soon

The problem is with reformatting multidimensional array data, the result is just awful.

jorenham commented:
> > It probably won't be enforced globally anytime soon
> 
> The problem is with reformatting multidimensional array data, the result is just awful.

Yea I agree. But in this case it's a 1d array that's arbitrarily split over two lines. 

BTW, enforcing `ruff format` isn't absolute: You can temporarily disable it with `# fmt: skip` for a single statement, and with `# fmt: off` and `# fmt: on` for a range of code.

DimitriPapadopoulos commented on numpy/_core/tests/test_datetime.py:
```diff
@@ -2492,7 +2492,7 @@ def test_isfinite_isinf_isnan_units(self, unit, dstr):
         '''
         arr_val = [123, -321, "NaT"]
-        arr = np.array(arr_val,  dtype= dstr % unit)
+        arr = np.array(arr_val, dtype=(dstr % unit))
```
Are you sure about removing the parenthesis? I find the overall result less readable. I have checked `ruff format` and none of the linter rules of `ruff check` care about the parentheses.

jorenham replied:
It's just that this isn't a `ruff` fix, as the PR title seems to suggest.

DimitriPapadopoulos replied:
Indeed, but ruff removed the space around `=` which I believe makes the result less readable without parentheses.

DimitriPapadopoulos committed:
STY: More taking into account reviewer's comments

Unwrap the array arbitrarily split over two lines, as it fits the new
default line-length limit of 88 characters.

Co-authored-by: Joren Hammudoglu <jhammudoglu@gmail.com>

DimitriPapadopoulos commented:
I unwrapped the split line as it fits the new line-length limit of 88 characters.

By the way, would you accept a PR that runs `ruff format` on Numpy, but wraps multidimensional arrays in `# fmt: off`/`on`?

jorenham commented:
> By the way, would you accept a PR that runs `ruff format` on Numpy, but wraps multidimensional arrays in `# fmt: off`/`on`?

Only if it's possible to enforce a different max-line-length for `.py` (88) and `.pyi` (130) (astral-sh/ruff#16057, astral-sh/ruff#7696).

charris commented:
Rebase needed.

jorenham commented:
Thanks Dimitri :)
